### PR TITLE
Feature build static gcc

### PR DESCRIPTION
**Pull request by sarfata**
Build GCC without external dependencies.

### CODE REVIEW

**Comment by martijnthe** [comment]
Looks good to me. @bradtgmurray ?

File: Makefile [hunk 1]
```diff
@@ -275,7 +277,9 @@ cross-gcc: cross-binutils cross-gcc-first cross-newlib gcc-$(GCC_VERSION)-$(CS_B
 	--disable-libstdcxx-pch	--enable-poison-system-directories 	\
 	--with-sysroot="$(PREFIX)/$(TARGET)"				\
 	--with-build-time-tools="$(PREFIX)/$(TARGET)/bin"		\
-	--enable-extra-sgxxlite-multilibs $(CS_SPECS) && \
+	--with-gmp=$(PREFIX) --with-mpfr=$(PREFIX) --with-mpc=$(PREFIX) \
```
Comment (bradtgmurray) [inline Makefile]: with-gmp and friends are here twice?

File: Makefile [hunk 2]
```diff
@@ -241,7 +242,7 @@ CS_SPECS='--with-specs=%{save-temps: -fverbose-asm}			\
 %{O*:%{O|O0|O1|O2|Os:;:%{!fno-remove-local-statics:			\
 -fremove-local-statics}}}'
 
-cross-gcc-first: cross-binutils gcc-$(GCC_VERSION)-$(CS_BASE) multilibbash 
+cross-gcc-first: gmp mpfr mpc cross-binutils gcc-$(GCC_VERSION)-$(CS_BASE) multilibbash
```
Comment (bradtgmurray) [inline Makefile]: As far as I know the changes to this target are unnecessary. gcc-first is only built as a build artifact, it's a compiler that's only built to build newlib and then it's used to rebuild gcc itself in the cross-gcc target. End users will only use the final one. However, it doesn't hurt...

**Comment by bradtgmurray** [comment]
:+1:

File: Makefile (hunk 2 above)
Comment (sarfata) [inline Makefile]: You need this to be able to build without the libs installed on your machine. I had removed them to make sure that it was built with few external deps. I think it makes sense to keep that in, it also simplifies the requirements if someone wants to rebuild the tools.

File: Makefile (hunk 1 above)
Comment (sarfata) [inline Makefile]: That's a mistake that careful diff examination should have caught. Shame on me :confounded:

**Comment by sarfata** [comment]
Not merging yet because I have a problem building this on ubuntu 13.04 ... I will try to get one single script that builds everywhere before I merge.

**Comment by sarfata** [comment]
Closing this because this was included in the incremental build PR of @Meiguro which is now merged.

**Pull request closed without merging**